crates/yas_periphery/src/yas_nft_position_manager.cairo

```diff
@@ -1,24 +1,635 @@
+            let msg_sender = get_caller_address();
+
+            // TODO: we need verify if data has a valid ContractAddress
+            let mut sender: ContractAddress = Zeroable::zero();
+            if !data.is_empty() {
+                sender = (*data[0]).try_into().unwrap();
+            }
+
+            self.emit(MintCallback { amount_0_owed, amount_1_owed });
+
+            if amount_0_owed > 0 {
+                let token_0 = IYASPoolDispatcher { contract_address: msg_sender }.token_0();
+                IERC20Dispatcher { contract_address: token_0 }
+                    .transferFrom(sender, msg_sender, amount_0_owed);
+            }
+            if amount_1_owed > 0 {
+                let token_1 = IYASPoolDispatcher { contract_address: msg_sender }.token_1();
+                IERC20Dispatcher { contract_address: token_1 }
+                    .transferFrom(sender, msg_sender, amount_1_owed);
+            }
+        }
+
+        // ERC721
+        fn supports_interface(self: @ContractState, interface_id: felt252) -> bool {
+            let state = ERC721::unsafe_new_contract_state();
+            ERC721::SRC5Impl::supports_interface(@state, interface_id)
+        }
+
+        fn name(self: @ContractState) -> felt252 {
+            let state = ERC721::unsafe_new_contract_state();
+            ERC721::ERC721MetadataImpl::name(@state)
+        }
+
+        fn symbol(self: @ContractState) -> felt252 {
+            let state = ERC721::unsafe_new_contract_state();
+            ERC721::ERC721MetadataImpl::symbol(@state)
+        }
+
+        fn token_uri(self: @ContractState, token_id: u256) -> felt252 {
+            let state = ERC721::unsafe_new_contract_state();
+            assert(ERC721::InternalImpl::_exists(@state, token_id), 'ERC721: invalid token ID');
+            // TODO: url
+            1
+        }
+
+        fn balance_of(self: @ContractState, account: ContractAddress) -> u256 {
+            let state = ERC721::unsafe_new_contract_state();
+            ERC721::ERC721Impl::balance_of(@state, account)
+        }
+
+        fn owner_of(self: @ContractState, token_id: u256) -> ContractAddress {
+            let state = ERC721::unsafe_new_contract_state();
+            ERC721::ERC721Impl::owner_of(@state, token_id)
+        }
+
+        fn transfer_from(
+            ref self: ContractState, from: ContractAddress, to: ContractAddress, token_id: u256
+        ) {
+            let mut state = ERC721::unsafe_new_contract_state();
+            ERC721::ERC721Impl::transfer_from(ref state, from, to, token_id);
+        }
+
+        fn safe_transfer_from(
+            ref self: ContractState,
+            from: ContractAddress,
+            to: ContractAddress,
+            token_id: u256,
+            data: Span<felt252>
+        ) {
+            let mut state = ERC721::unsafe_new_contract_state();
+            ERC721::ERC721Impl::safe_transfer_from(ref state, from, to, token_id, data);
+        }
+
+        // TODO: replace _approve
```

Comment:
Why is this necessary?

Reply:
You're right, the comment is not necessary. I've already removed it

Reply:
✔️